src/odoc/bin/main.ml

```diff
@@ -187,14 +192,34 @@ end = struct
               "Either --package or --parent should be specified, not both")
     in
     let source =
-      match (source_parent_file, source_name) with
-      | Some parent, Some name -> Ok (Some (parent, name, source_cmt))
-      | Some _, None | None, Some _ ->
+      match
+        (source_parent_file, source_name, source_cmt, Fs.File.get_ext input)
+      with
+      | Some parent, Some name, None, ".cmt" -> Ok (Some (parent, name, input))
+      | Some parent, Some name, Some cmt, ".cmt" ->
+          if Fpath.equal cmt input then Ok (Some (parent, name, input))
+          else
+            Error
+              (`Cli_error
+                "--cmt has to be equal to the input file when this one has \
+                 .cmt extension.")
+      | Some parent, Some name, Some cmt, _ -> Ok (Some (parent, name, cmt))
+      | Some _, Some _, None, _ ->
+          Error
+            (`Cli_error
+              "--cmt has to be passed when --source-parent-file and \
+               --source-name are passed and the input file is not a cmt file.")
```

Comment:
Shouldn't we handle the case where you want to render source but we don't have the cmt?

Reply:
We can, in which case there will be no jump to location, and if I'm not mistaken, no way to make a link to it.
It will still be accessible by "hardcoding" the link to it.